tket/src/Transformations/Decomposition.cpp

```diff
@@ -616,7 +617,14 @@ static double best_noise_aware_decomposition(
     unsigned max_nzz = fid.ZZMax_fidelity ? 1 : 3;
     for (unsigned n_zz = 0; n_zz <= max_nzz; ++n_zz) {
       if (n_zz > 0) {
-        double gate_fid = (*fid.ZZPhase_fidelity)(angles[n_zz - 1]);
+        double gate_fid = std::visit(
+            overloaded{// Constant value
+                       [](double arg) { return arg; },
+                       // A value depending on the angle
+                       [angles, n_zz](std::function<double(double)> arg) {
```

Comment:
Not sure about capturing `angles` here in the lambda, possibly could be capturing it by reference?

Reply:
apparently it's a size 3 array, so probably safe